dpctl/tensor/_dlpack.pyx

```diff
@@ -291,14 +296,29 @@ cpdef to_dlpack_capsule(usm_ndarray usm_ary):
         for i in range(nd):
             shape_strides_ptr[i] = shape_ptr[i]
         strides_ptr = usm_ary.get_strides()
+        flags = usm_ary.flags_
         if strides_ptr:
             for i in range(nd):
                 shape_strides_ptr[nd + i] = strides_ptr[i]
         else:
-            si = 1
-            for i in range(0, nd):
-                shape_strides_ptr[nd + i] = si
-                si = si * shape_ptr[i]
+            if flags & USM_ARRAY_C_CONTIGUOUS:
+                si = 1
+                for i in range(nd - 1, -1, -1):
+                    shape_strides_ptr[nd + i] = si
+                    si = si * shape_ptr[i]
+            elif flags & USM_ARRAY_F_CONTIGUOUS:
+                si = 1
+                for i in range(0, nd):
+                    shape_strides_ptr[nd + i] = si
+                    si = si * shape_ptr[i]
+            else:
+                stdlib.free(shape_strides_ptr)
+                stdlib.free(dlm_tensor)
+                raise BufferError(
+                    "to_dlpack_capsule: Could not reconstruct strides "
+                    "for non-contiguous tensor"
+                )
```

Comment:
good that we're covering this case, I think it would be better to be more specific about the problem though

in `dpctl4pybind11.hpp` we throw
```
  throw std::runtime_error("Invalid array encountered when "
                           "building strides");
```
something like this might make more sense, so it's understood the array is invalid

Reply:
Make sense, thank you!